doc/developer_guide/psyir_symbols.rst

```diff
@@ -204,6 +204,11 @@ the current implementation so that either a) references went in both
 directions or b) references were replaced with names and lookups. Each
 of these solutions has their benefits and disadvantages.
 
+A third solution would be for Symbol to be non-hierarchical class with just
```

Comment:
I rephrased following your suggestion. The idea would be an reference box with the name and the symbol-type attribute, this would be the RoutineSymbol, DataSymbol etc (maybe with postfixed with Type as they are not Symbol classes anymore), so this can be mutated without changing the overall reference box. Note that this type is not the same as the data-type, which may be confusing.